### ISSUE TITLE
boot.py definitions lost when using Stop/Restart backend button

### ISSUE THREAD

**Issue by adzierzanowski** [thonny/thonny#976]
Consider a simple micropython `boot.py` script:

```python
import time

hello = 'hello, world'

while True:
    print(hello)
    time.sleep(1)
```

When interrupting within the shell with ctrl+C, it just throws an exception and I still can use the `hello` variable in the interpreter. But when I use the Stop button instead, the backend restarts and the access to `hello` is lost because it is no longer defined although it should be because `boot.py` runs everytime the MCU is restarted.

**Comment by aivarannamaa** [comment]
Thank you for the report!

I wanted to emulate the behavior of CPython backend, where Stop/Restart really cleans the environment (by restarting the process and not running anything).

In MicroPython back-end I clear the environment manually, but it looks like this approach doesn't suit here.

**Comment by aivarannamaa** [comment]
Fixed in 3.3
